### PR TITLE
#276 RsWithStatus.java:97-101: To implement concatenation and selection

### PR DESCRIPTION
**Pull request by dmzaytsev**
For #276 

*  implemented concatenation and selection with the conjunction Concat and Select class to get rid of List.add() according the puzzle description 
* created unit test for `RsWithStatus`

### CODE REVIEW

**Comment by davvd** [comment]
@dmzaytsev Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@pinaf it's yours, please review

**Comment by dmzaytsev** [comment]
@pinaf take a look this PR please

File: src/test/java/org/takes/rs/RsWithStatusTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import com.google.common.base.Joiner;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link RsWithStatus}..
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithStatusTest.java]: @dmzaytsev extra dot here - typo

**Comment by pinaf** [comment]
@dmzaytsev looks good. just 1 minor comment.

**Comment by dmzaytsev** [comment]
@pinaf fixed

**Comment by pinaf** [comment]
@rultor merge

File: src/main/java/org/takes/rs/RsWithStatus.java [hunk 2]
```diff
@@ -117,17 +119,20 @@ public InputStream body() throws IOException {
...
+        return new Concat<String>(
+            Collections.singletonList(
+                String.format("HTTP/1.1 %d %s", status, reason)
+            ),
+            new Select<String>(
+                origin.head(),
+                new Condition<String>() {
+                    @Override
+                    public boolean fits(final String element) {
+                        return !RsWithStatus.FIRST.matcher(element).matches();
```
Comment (yegor256) [inline src/main/java/org/takes/rs/RsWithStatus.java]: @dmzaytsev eghh... I don't like. you're checking every single line for a regex matching.. this is 1) slow and 2) redundant. all you need to do is to filter out the first line. I would use an `AtomicBoolean` flag for that

**Comment by yegor256** [comment]
@dmzaytsev see my comment above

**Comment by dmzaytsev** [comment]
@yegor256 fixed, looks good now ?

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@rultor please deploy

**Comment by davvd** [comment]
@pinaf Thanks for your contribution, **17 mins** was added to your account, payment ID is `AP-5HJ495614K352512V`, 70 hours and 49 mins spent total. you're getting extra minutes here (c=2). +17 added to your rating, current score is: [+7623](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**